MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm

```diff
@@ -334,7 +334,17 @@
 	initDebugCallbacks(pCreateInfo);	// Do before any creation activities
 
 	mvkSetOrClear(&_appInfo, pCreateInfo->pApplicationInfo);
-	if (_appInfo.apiVersion == 0) { _appInfo.apiVersion = VK_API_VERSION_1_0; }	// Default
+    
+    if (_appInfo.apiVersion == 0) {
+        _appInfo.apiVersion = VK_API_VERSION_1_0;   // Default
+    }
+    else if (VK_API_VERSION_MAJOR(_appInfo.apiVersion) > MVK_VERSION_MAJOR ||
+             VK_API_VERSION_MINOR(_appInfo.apiVersion) > MVK_VERSION_MINOR) {
```

Comment:
This feels wrong. Probably because it doesn't handle the case where `apiVersion` is nonzero but `VK_API_VERSION_MAJOR(apiVersion)` is zero--even though that shouldn't happen in practice.

It should just be possible to compare `_appInfo.apiVersion` with `MVK_VULKAN_API_VERSION` anyway. That includes the header revision. That way, if an app specifies an `apiVersion` with a greater header revision, the warning will be generated, which would serve as a good reminder to upgrade MoltenVK.

Reply:
Yeah, I agree with you, we also need to handle incorrect apiVersion. What warning message do you think is better to display?